src/api/common/Env.ts

```diff
@@ -186,9 +150,25 @@ export function isOfflineStorageAvailable(): boolean {
 	return !isBrowser()
 }
 
-export function assertOfflineStorageAvailable() {
-	if (!isOfflineStorageAvailable()) {
-		throw new Error("Offline storage is not available")
+export class DomainConfigProvider {
```

Comment:
i think this should be a separate file, it  also feels from the usages that i could wrap our getWebRoot() and getApiUrl() and similar functions and hide the domain config completely? Basically make this an abstraction for anytime anyone wants to know something that depends on the current origin

Reply:
The abstraction is that we don't touch `DomainConfig` directly but always use the accessors?
I didn't know how to do things without building too many abstractions